lighthouse-core/report/html/renderer/performance-category-renderer.js

```diff
@@ -298,6 +309,68 @@ class PerformanceCategoryRenderer extends CategoryRenderer {
     element.appendChild(passedElem);
     return element;
   }
+
+  /**
+   * Render the control to filter the audits by metric. The filtering is done at runtime by CSS only
+   * @param {LH.ReportResult.AuditRef[]} filterableMetrics
+   * @param {HTMLDivElement} element
+   */
+  renderMetricAuditFilter(filterableMetrics, element) {
+    filterableMetrics.unshift(/** @type {LH.ReportResult.AuditRef} */ ({acronym: 'All'}));
+    // thx https://codepen.io/surjithctly/pen/weEJvX
+    const filterBarEl = this.dom.createElement('div', 'lh-filterbar');
+    const filterBarSummaryEl = this.dom.createChildOf(filterBarEl, 'span', 'lh-filterbar__summary');
+    filterBarSummaryEl.textContent = 'Filter to relevant audits: ';
+    const labelSelectors = [];
+    const auditSelectors = [];
+    for (const metric of filterableMetrics) {
+      // appended into `element` for sweet ~ selectors to work
+      const elemId = `metric-${metric.acronym}`;
+      const radioEl = this.dom.createChildOf(element, 'input', 'lh-filterbar__radio', {
+        type: 'radio',
+        name: 'metricsfilter',
+        id: elemId,
+        hidden: 'true',
+      });
+      const labelEl = this.dom.createChildOf(filterBarEl, 'label', 'lh-filterbar__label', {
+        for: elemId,
+      });
+      labelEl.textContent = metric.acronym || metric.id;
+      if (metric.acronym === 'All') {
+        radioEl.checked = true;
+      }
+      // Dynamically write some CSS, for the CSS-only filtering
```

Comment:
all this dynamic CSS seems much harder to follow than the equivalent JS would be. What reasons did you rule out JS?